Sources/MenuAnimator.swift

```diff
@@ -231,86 +87,260 @@ class MenuInteractiveTransition: NSObject, UIViewControllerInteractiveTransition
+        guard let toViewController = transitionContext.viewController(forKey: .to) else {
             fatalError("Invalid toViewController key. Can't start transition")
         }
         let containerView = transitionContext.containerView
-        
         let screenWidth = containerView.frame.size.width
         
         if present {
             containerView.insertSubview(toViewController.view, belowSubview: fromViewController.view)
-
-            if self.tapRecognizer == nil {
-
+            
+            if tapRecognizer == nil {
+                
                 guard toViewController is MenuViewController else {
                     fatalError("Invalid toViewController type. It must be MenuViewController")
                 }
-                self.tapRecognizer = UITapGestureRecognizer(target: toViewController,
-                                                            action: #selector(MenuViewController.handleTap(recognizer:)))
-                self.panRecognizer = UIPanGestureRecognizer(target: self,
-                                                        action: #selector(MenuInteractiveTransition.handlePanDismission(recognizer:)))
+                tapRecognizer = UITapGestureRecognizer(target: toViewController,
+                                                       action: #selector(MenuViewController.handleTap(recognizer:)))
+                panRecognizer = UIPanGestureRecognizer(target: self,
+                                                       action: #selector(MenuInteractiveTransition.handlePanDismission(recognizer:)))
             }
-
+            
             contentSnapshotView = createSnapshotView(from: fromViewController.view)
-
+            
             guard let contentSnapshotView = self.contentSnapshotView else {
                 fatalError("Invalid `contentSnapshotView` value. This property should not be nil")
             }
-
+            
             containerView.addSubview(contentSnapshotView)
-
+            
             fromViewController.view.isHidden = true
         } else {
             containerView.addSubview(toViewController.view)
-
+            
             toViewController.view.transform = CGAffineTransform(scaleX: options.contentScale, y: options.contentScale)
-            addShadow(toView: toViewController.view)
-
+            addShadow(to: toViewController.view)
+            
             let newOrigin = CGPoint(x: screenWidth - options.visibleContentWidth, y: toViewController.view.frame.origin.y)
             let rect = CGRect(origin: newOrigin, size: toViewController.view.frame.size)
-
+            
             toViewController.view.frame = rect
         }
-
+        
         toViewController.view.isUserInteractionEnabled = false
         fromViewController.view.isUserInteractionEnabled = false
     }
-
+    
+    
+    fileprivate func finishTransition(currentPercentComplete : CGFloat) {
```

Comment:
Please, move two `fileprivate` methods before all private methods, and just after public ones.